src/base/table.lua

```diff
@@ -373,18 +373,22 @@
 		local result = {}
 		local arg = {...}
 		for _,t in ipairs(arg) do
-
 			if type(t) == "table" then
-				for k,v in pairs(t) do
-					if type(result[k]) == "table" and type(v) == "table" then
-						result[k] = table.merge(result[k], v)
-					else
-						result[k] = v
+				if #t > 0 then  -- Array-based table
```

Comment:
Not necessary the correct check table (probably not from premake) might look like `local t ={a="b", [1]=42, [2]=21}`
unsure of the expected result with that kind of table though.